src-tauri/src/main.rs

```diff
@@ -628,90 +590,58 @@ fn load_encounters_preview(
         .resource_dir()
         .expect("could not get resource dir");
     let conn = get_db_connection(&path).expect("could not get db connection");
-
-    let min_duration = filter.min_duration * 1000;
-
-    let mut params = vec![min_duration.to_string()];
-
-    let search_words: Vec<&str> = if search.chars().any(|c| !c.is_whitespace()) {
-        search.split_whitespace().collect()
+    let mut params = vec![];
+
+    let join_clause = if search.len() > 2 {
+        let escaped_search = search
+            .split_whitespace()
+            .map(|word| format!("\"{}\"", word.replace("\"", "")))
+            .collect::<Vec<_>>()
+            .join(" ");
+        params.push(escaped_search);
+        "JOIN encounter_search(?) ON encounter_search.rowid = e.id"
     } else {
-        vec![""]
+        ""
     };
 
-    params.extend(
-        search_words
-            .iter()
-            .flat_map(|word| std::iter::repeat(word.to_string()).take(3)),
-    );
-
-    let word_count = search_words.len();
-
-    let join_clauses = (0..word_count).fold(String::new(), |acc, i| {
-        acc + &format!("JOIN entity ent{} ON e.id = ent{}.encounter_id\n    ", i, i)
-    });
-
-    let input_filter = (0..word_count)
-    .fold(String::new(), |acc, i| {
-        acc + &format!(
-            "AND ((current_boss LIKE '%' || ? || '%') OR (ent{}.class LIKE '%' || ? || '%') OR (ent{}.name LIKE '%' || ? || '%'))\n    ",
-            i, i
-        )
-    });
+    params.push((filter.min_duration * 1000).to_string());
 
     let boss_filter = if !filter.bosses.is_empty() {
-        let placeholders: Vec<String> = filter.bosses.iter().map(|_| "?".to_string()).collect();
-        filter.bosses.into_iter().for_each(|boss| params.push(boss));
-        format!("AND (current_boss IN ({}))", placeholders.join(","))
-    } else {
-        "".to_string()
-    };
-
-    let class_filter = if !filter.classes.is_empty() {
-        let placeholders: Vec<String> = filter.classes.iter().map(|_| "?".to_string()).collect();
-        filter
-            .classes
-            .into_iter()
-            .for_each(|class| params.push(class));
-        format!("AND (class IN ({}))", placeholders.join(","))
+        let mut placeholders = "?,".repeat(filter.bosses.len());
+        placeholders.pop(); // remove trailing comma
+        params.extend(filter.bosses);
+        &format!("AND e.current_boss IN ({})", placeholders)
```

Comment:
i have an error with the borrow checker here

you can just return 

`format!("AND e.current_boss IN ({})", placeholders)`

and in the else branch return

`"".to_string()`